wordpress-mobile/WordPress-iOS
# UL&S: Error label displays

Fixes # n/a
Auth PR: https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/pull/312

This PR connects the Continue button behavior and the error label table view cell to the LoginViewController's error message behavior.

**I did not include two of the old error handling states. These are:**
1. A method frequently used, `refreshSiteAddressError(immediate:)`. It was effectively dead code. The `showError` bool will always return `false` because the method `!loginFields.validateSiteForSignin()` always returns `false`. I did get the method working correctly again, only to find out that the user experience with this method working correctly is terribly annoying. It validates the URL as you type, so an error message immediately displays.
2. The `handleTextFieldDidChange(_ sender:)` method no longer immediately wipes out displayed errors. The old behavior immediately deletes an error message after changing a single letter or deleting a single letter in the textfield. I tried incorporating this behavior and timeboxed it to 4 hours. Since the error label now lives in a tableViewCell, it reloads the table data and dismisses the keyboard after a single change to the URL, after an error displayed. That behavior was annoying, and I didn't get the correct state, UI display, nor the correct behavior after 4 hours, so I didn't commit that. 

**Known bug**
There is something strange going on with the tableView scroll after an error message appears. The tableView allows you to scroll up slightly, but then it gets stuck and won't respond to touches to scroll back down. https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/issues/318

### To test:
1. Enable the .siteAddress Feature Flag
2. `rake dependencies`
3. build and run
4. Log In > Or log in by entering your site address
5. Enter a valid, non-WordPress URL, such as google.com, then Continue
6. **Expected:** an error message label is displayed in the table
7. Enter a valid WordPress URL, such as thuycopeland.com, then Continue
8. **Expected:** the error label is erased while it validates the URL. After it validates, it navigates you to the username / password screen.

| Initial state | Error state
| --- | --- |
| <a href="https://user-images.githubusercontent.com/1062444/86845973-da06c100-c06f-11ea-8bc4-2360372b87b8.png"><img src="https://user-images.githubusercontent.com/1062444/86845973-da06c100-c06f-11ea-8bc4-2360372b87b8.png" width="350" /></a> | <a href="https://user-images.githubusercontent.com/1062444/86845999-e68b1980-c06f-11ea-9894-c164644b00ef.png"><img src="https://user-images.githubusercontent.com/1062444/86845999-e68b1980-c06f-11ea-9894-c164644b00ef.png" width="350" /></a> |
| <a href="https://user-images.githubusercontent.com/1062444/86847481-00c5f700-c072-11ea-8b9f-66807eae6906.png"><img src="https://user-images.githubusercontent.com/1062444/86847481-00c5f700-c072-11ea-8b9f-66807eae6906.png" width="350" /></a> | |

PR submission checklist:
